src/main/java/common/Carrier.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * 
+ */
+package common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This abstract class defines a Carrier that will have concrete implementations
+ * for different RAT types.
+ */
+public abstract class Carrier {
+	protected int carrierId;
+	protected List<RfPort> rfPorts;
+	protected CarrierFrequency carrierFreq;
+	protected double transmittingPower;
+	
+	public Carrier(int carrierId, List<RfPort> rfPorts, CarrierFrequency freq)
+	{
+		this.carrierId = carrierId;
+		this.carrierFreq = freq;
+		
+		if (rfPorts == null || rfPorts.isEmpty())
+		{
+			System.out.println("ERROR: attempting create carrier without RF ports");
+		}
+		
```

Comment:
Just to check, would we still want to create a carrier without RF ports? Or should we break out of the method?

Reply:
I get the point here, and to me, there's nothing in the requirements that states a carrier cannot be instantiated with RF ports. But, we can also be lazy since it isn't explicitly stated, and I think we may be able to make the decision to handle it how we would want to. 
In this case, I think it saves us some trouble to just error out here and prevent instantiating incomplete carriers that could (potentially) screw up something with the RUs.